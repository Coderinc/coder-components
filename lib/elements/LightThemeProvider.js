import React from 'react';
import { ThemeProvider } from 'styled-components';
import lightThemeBase from '../styles/lightTheme';
import baseColors from '../styles/baseColors';

export default props => {
  const { primary = 'pink' } = props;

  const theme = {};
  Object.keys(lightThemeBase).forEach(k => {
    theme[k] = baseColors[lightThemeBase[k]] || lightThemeBase[k];
  });
  theme.primary = baseColors[primary];

  return (
    <ThemeProvider theme={theme}>
      {props.children}
    </ThemeProvider>
  )
};
