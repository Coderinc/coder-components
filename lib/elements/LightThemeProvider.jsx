import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from '../../node_modules/styled-components';
import lightThemeBase from '../styles/themes/lightTheme';
import baseColors from '../styles/base-colors';

const LightThemeProvider = ({ primary, children }) => {
  const theme = {};
  Object.keys(lightThemeBase).forEach((k) => {
    theme[k] = baseColors[lightThemeBase[k]] || lightThemeBase[k];
  });
  theme.primary = baseColors[primary];

  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
};

LightThemeProvider.propTypes = {
  primary: PropTypes.string,
  children: PropTypes.element,
};

LightThemeProvider.defaultProps = {
  primary: 'pink',
  children: null,
};

export default LightThemeProvider;
