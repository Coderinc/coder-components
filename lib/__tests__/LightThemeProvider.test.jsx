import React from 'react';
import renderer from 'react-test-renderer';
import LightThemeProvider from '../elements/LightThemeProvider';

describe('LightThemeProvider', () => {
  it('renders correctly', () => {
    const component = renderer.create(<LightThemeProvider />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
