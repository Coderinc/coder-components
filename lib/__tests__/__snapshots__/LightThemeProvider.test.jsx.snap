// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightThemeProvider renders correctly 1`] = `null`;
